### PR TITLE
WeChat login in react native

### PR DESCRIPTION
**Pull request by carmenlau**
refs #52 

Example: https://github.com/carmenlau/authgear-sdk-js/tree/52-wechat-login-react-native-example

### CODE REVIEW

File: packages/authgear-react-native/src/eventEmitter.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  NativeModules,
+  Platform,
+  NativeEventEmitter,
+  DeviceEventEmitter,
+  EventEmitter,
+} from "react-native";
+
+export default Platform.select<EventEmitter>({
+  ios: new NativeEventEmitter(NativeModules.AuthgearManager),
+  android: DeviceEventEmitter,
```
Comment (louischan-oursky) [inline packages/authgear-react-native/src/eventEmitter.ts:11]: I just wonder if we could use the same approach (NativeEventEmitter) on android as well.

File: packages/authgear-react-native/src/index.ts [hunk 2]
```diff
@@ -163,10 +166,15 @@ export class ReactNativeContainer<
...
     });
+    const deepLinkListener = (url: string) => {
+      this._handleWeChatRedirectURI(url, options.weChatRedirectURI);
+    };
+    EventEmitter.addListener("onAuthgearDeepLink", deepLinkListener);
     const redirectURL = await openAuthorizeURL(
       authorizeURL,
       options.redirectURI
     );
+    EventEmitter.removeListener("onAuthgearDeepLink", deepLinkListener);
```
Comment (louischan-oursky) [inline packages/authgear-react-native/src/index.ts:177]: The remove statement should be put in the finally block.

File: packages/authgear-react-native/src/eventEmitter.ts (hunk 1 above)
Comment (carmenlau) [reply packages/authgear-react-native/src/eventEmitter.ts]: Updated in 0bf3d99, finally found that we only need to pass undefined to `NativeEventEmitter` for non ios platform.

ref: https://github.com/facebook/react-native/blob/a715c5eba7ed7a7d8da22ba833ff135392255ba5/Libraries/Linking/Linking.js#L33

**Comment by louischan-oursky** [comment]
@carmenlau conflicted.

**Comment by carmenlau** [comment]
Updated.

**Pull request merged**